src/vanilla/ClientModelExtensions.cs

```diff
@@ -853,53 +853,27 @@ public static string ConstructMapper(this IModelType type, string serializedName
                     builder.AppendLine("className: '{0}',", composite.Name)
                            .AppendLine("modelProperties: {").Indent();
                     var composedPropertyList = new List<Property>(composite.ComposedProperties);
+                    PropertyInfo itemName = composite.GetType().GetProperty("ItemName");
+                    string itemNameValue = (string)itemName?.GetValue(composite);
+                    PropertyInfo nextLinkName = composite.GetType().GetProperty("NextLinkName");
+                    bool writeAllModelProperties = !isPageable || nextLinkName == null || (string)nextLinkName.GetValue(composite) != null;
                     for (var i = 0; i < composedPropertyList.Count; i++)
                     {
-                        var prop = composedPropertyList[i];
-                        var serializedPropertyName = prop.SerializedName;
-                        PropertyInfo nextLinkName = null;
-                        string nextLinkNameValue = null;
-                        if (isPageable)
+                        Property prop = composedPropertyList[i];
+
+                        if (ShouldWriteProperty(writeAllModelProperties, prop))

```

Comment:
Now we're adding a special case where sometimes we don't write a nextLink property?

Reply:
That case always existed, it was just buried a little bit in the code. The real thing we're doing here is changing the logic that determines whether a comma is added to the end of the property's object value or not. Previously we did it by detecting what type of object was being written, but I changed it to just detect whether we're going to write the next model property or not.